packages/itwinui-react/src/core/ComboBox/ComboBoxMenu.tsx

```diff
@@ -49,18 +58,42 @@ const VirtualizedComboBoxMenu = (props: React.ComponentProps<'div'>) => {
     );
   }, [focusedIndex, menuRef]);
 
-  const { outerProps, innerProps, visibleChildren } = useVirtualization({
-    // 'Fool' VirtualScroll by passing length 1
-    // whenever there is no elements, to show empty state message
-    itemsLength: filteredOptions.length || 1,
-    itemRenderer: virtualItemRenderer,
-    scrollToIndex: focusedVisibleIndex,
-  });
+  useLayoutEffect(() => {
+    if (focusedVisibleIndex) {
+      virtualizer.scrollToIndex(focusedVisibleIndex, { align: 'start' });
+    }
+  }, [focusedVisibleIndex, virtualizer]);
+
+  const innerProps = {
+    style: { willChange: 'transform' },
+  } as const;
 
   return (
-    <Box as='div' {...outerProps} {...rest}>
-      <div {...innerProps} ref={innerProps.ref}>
-        {visibleChildren}
+    <Box
+      as='div'
+      style={{
+        minBlockSize: virtualizer.getTotalSize(),
+        minInlineSize: '100%',
+      }}
+      {...rest}
+    >
+      <div {...innerProps}>
+        {virtualizer.getVirtualItems().map((virtualItem) => (
+          <div
```

Comment:
I don't think there should be an extra div needed here. All these props could be easily applied on the items directly.

Reply:
Removed the wrapping div in this [commit](https://github.com/iTwin/iTwinUI/pull/2061/commits/89a25d2099cb669a376560aff8e61a42b45d6c3f), and also removed the code accounting for it from the keyboard navigation handling. Also removed the mention of change to the keyboard handling from the PR description.